pkg/aws/ec2/api/helper.go

```diff
@@ -434,77 +439,120 @@ func (h *ec2APIHelper) GetInstanceDetails(instanceId *string) (*ec2.Instance, er
+		len(assignPrivateIPOutput.AssignedIpv4Prefixes) == 0) {
+		return assignedResources, fmt.Errorf("failed to create %v %s to eni %s", count, resourceType, eniID)
 	}
 
-	ErrIPNotAttachedYet := fmt.Errorf("private IPv4 address is not attached yet")
+	ErrIPResourceNotAttachedYet := fmt.Errorf("%v is not attached yet", resourceType)
 
 	err = retry.OnError(waitForIPAttachment,
 		func(err error) bool {
-			if err == ErrIPNotAttachedYet {
-				// Retry in case IPs are not attached yet
+			if err == ErrIPResourceNotAttachedYet {
+				// Retry in case IPv4 Resources are not attached yet
 				return true
 			}
 			return false
 		}, func() error {
-			// Describe the network interface on which the new IPs are assigned
+			// Describe the network interface on which the new IP or prefixes are assigned
 			interfaces, err := h.DescribeNetworkInterfaces([]*string{&eniID})
-			// Re initialize the slice so we don't add IPs multiple time
-			assignedIPs = []string{}
-			if err == nil && len(interfaces) == 1 && interfaces[0].PrivateIpAddresses != nil {
-				// Get the map of IPs returned by the describe network interface call
-				ipAddress := map[string]bool{}
-				for _, ipAddr := range interfaces[0].PrivateIpAddresses {
-					ipAddress[*ipAddr.PrivateIpAddress] = true
-				}
-				// Verify the describe network interface returns all the IPs that were assigned in the
-				// AssignPrivateIPAddresses call
-				for _, ip := range assignPrivateIPOutput.AssignedPrivateIpAddresses {
-					if _, ok := ipAddress[*ip.PrivateIpAddress]; !ok {
-						// Even if one IP is not assigned, set the error so that we only return only the IPs that
-						// are successfully assigned on the ENI
-						err = ErrIPNotAttachedYet
-					} else {
-						assignedIPs = append(assignedIPs, *ip.PrivateIpAddress)
+			// Re-initialize the slice so that we don't add IP resources multiple times
+			assignedResources = []string{}
+
+			if err == nil && len(interfaces) == 1 {
+				if resourceType == config.ResourceTypeIPv4Address && interfaces[0].PrivateIpAddresses != nil {
+					// Get the map of IPs returned by the describe network interface call
+					ipAddress := map[string]bool{}
+					for _, ipAddr := range interfaces[0].PrivateIpAddresses {
+						ipAddress[*ipAddr.PrivateIpAddress] = true
+					}
+					// Verify describe network interface returns all the IPs that were assigned in the
+					// AssignPrivateIPAddresses call
+					for _, ip := range assignPrivateIPOutput.AssignedPrivateIpAddresses {
+						if _, ok := ipAddress[*ip.PrivateIpAddress]; !ok {
+							// Even if one IP is not assigned, set the error so that we only return only the IPs that
+							// are successfully assigned on the ENI
+							err = ErrIPResourceNotAttachedYet
+						} else {
+							assignedResources = append(assignedResources, *ip.PrivateIpAddress)
+						}
+					}
+					return err
+				} else {
```

Comment:
nit: probably using `else if` can help readability and future more cases?

Reply:
Yea of course, updated.